spec/formatting.md

```diff
@@ -18,3 +18,212 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Error Handling
+
+During the formatting of a message,
+various errors may be encountered.
+These are divided into the following categories:
+
+- **Syntax errors** occur when the syntax representation of a message is not well-formed.
+
+  Example invalid messages resulting in a Syntax error:
+
+  ```
+  {Missing end brace
+  ```
+
+  ```
+  {Unknown {?placeholder}}
+  ```
+
+  ```
+  let $var = {(no message body)}
```

Comment:
No, the `let` line is fine; it's there just to indicate that the source isn't completely empty. The failure here is from there being nothing after the `let`, i.e. no Pattern or Selector+Variants.

Reply:
The empty pattern is not an error, though, right? It's not a very useful message, but it's not an error, is it?

Reply:
Yes, it's an error because it's missing the `{}` wrappers. Something like this would be a valid empty message:

```
let $var = {(no message body)}
{}
```

Reply:
So you're saying that the empty string is not a valid pattern and results in an error? That is, the minimum a pattern string can ever be is `{}`? That makes me uncomfortable....

Reply:
That is what the current EBNF states.